inference-engine/src/low_precision_transformations/src/network_helper.cpp

```diff
@@ -1568,36 +1568,32 @@ NetworkHelper::InsertDequantizationResult NetworkHelper::moveDequantizationAfter
 
     assert(operation->get_output_size() == 1);
 
+    // we must have dequantization multiply
+    assert(dequantization.multiply != nullptr);
+
     OutputVector inputs = operation->input_values();
     const size_t dequantizationIndex = getChildInputIndex(dequantization.multiply, operation);
-    inputs[dequantizationIndex] = moveSubtract ?
-        dequantization.data :
-        (dequantization.subtract == nullptr ? dequantization.data : dequantization.subtract);
+    inputs[dequantizationIndex] = (!moveSubtract && dequantization.subtract != nullptr) ?
+        dequantization.subtract :
+        dequantization.data;
 
-    const std::shared_ptr<ngraph::Node> newOperation = operation->clone_with_new_inputs(inputs);
+    const auto newOperation = operation->clone_with_new_inputs(inputs);
     newOperation->set_friendly_name(operation->get_friendly_name());
     ngraph::copy_runtime_info(operation, newOperation);
 
-    auto op = std::dynamic_pointer_cast<ngraph::op::TypeRelaxedBase>(newOperation);
-    if (op != nullptr) {
-        if (updatePrecision) {
-            op->set_overridden_output_type(newOperation->get_input_element_type(0));
-        } else if (dequantization.multiply) {
-            op->set_overridden_output_type(dequantization.multiplyConstant->get_element_type());
-        } else if (dequantization.subtract) {
-            op->set_overridden_output_type(dequantization.subtractConstant->get_element_type());
-        }
-        std::dynamic_pointer_cast<ngraph::Node>(newOperation)->validate_and_infer_types();
+    if (const auto op = std::dynamic_pointer_cast<ngraph::op::TypeRelaxedBase>(newOperation)) {
+        op->set_overridden_output_type(updatePrecision ?
+            newOperation->get_input_element_type(0) :
+            dequantization.multiplyConstant->get_element_type());
+        newOperation->validate_and_infer_types();
```

Comment:
@vzinovie, can you take a look here, please? Your changes from here https://github.com/openvinotoolkit/openvino/commit/23808f46f7b5d464fd649ad278f253eec12721b3 are are overrode.